### ISSUE TITLE
W0236: Invalid-overridden-method checker doesn't detect return type mismatches

### ISSUE THREAD

**Issue by Julfried** [pylint-dev/pylint#10351]
### Bug description

I recently stumbled across this in a project. The invalid-overridden-method (W0236) checker doesn't detect return type mismatches in overridden methods. Tools like Pyright/Pylance correctly detect this issue, but Pylint misses it.  

Here is an example code snippet:
```python
from abc import ABC, abstractmethod
from io import TextIOWrapper, BytesIO

class BaseClass(ABC):
    @abstractmethod
    def read_file(self, path: str) -> TextIOWrapper:
        """Abstract method that should return a TextIOWrapper."""
        raise NotImplementedError("Method must be implemented by subclass")

class ChildClass(BaseClass):
    def read_file(self, path: str) -> BytesIO:  # Should trigger invalid-overridden-method
        """Implementation returns BytesIO instead of TextIOWrapper."""
        return BytesIO(b"content")
```

### Command used

```shell
pylint my_module.py
```

### Pylint output

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)
```

### Expected behavior

Pylint should report an invalid-overridden-method (W0236) error since the return type BytesIO is not compatible with the parent method's return type TextIOWrapper.

Pyright correctly reports: "Method 'read_file' overrides class 'BaseClass' in an incompatible manner. Return type mismatch: base method returns type 'TextIOWrapper', override returns type 'BytesIO'."

### Pylint version

```shell
pylint 4.0.0-dev0
astroid 4.0.0a0
Python 3.13.2 (main, Mar 17 2025, 21:04:07) [MSC v.1943 64 bit (AMD64)]
```

**Comment by pavan-msys** [comment]
Hi @Julfried 

Can I work on this issue?

**Comment by pavan-msys** [comment]
Hi @Julfried 

I have worked on the issue and made changes to the codebase. After running pylint on the file, I received the following output:

```
(venv) msys@msys-virtual-machine:~/pylint$ pylint my_module.py
************* Module my_module
my_module.py:11:4: W0236: Method 'read_file' was expected to be <ClassDef.TextIOWrapper l.0 at 0x748f51aa0f10>, found it instead as <ClassDef.BytesIO l.0 at 0x748f51c06210> (invalid-overridden-method)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 0.00/10, +8.75)
```


**Could you please confirm if this is the expected output, or if I need to make any changes?**

**Comment by Julfried** [comment]
Yes thank you! Thats what I meant. Feel free to open a PR on that one :)

**Comment by Julfried** [comment]
No Problem, you can open a PR and get guidance on how to tackle the tests :)

**Comment by pavan-msys** [comment]
Hi @Julfried 

I have raised a PR. Could you please review the code and let me know if any changes are required?

Thanks!

**Comment by Julfried** [comment]
Thanks, unfortunately I am not a maintainer, so we will have to wait and see :)

**Comment by Julfried** [comment]
Closed as not planned #10449
